bootstrap.py

```diff
@@ -587,14 +580,44 @@ def install_foreman_ssh_key():
     except urllib2.URLError, exception:
         print_generic("Could not reach the server. Error: %s" % exception.reason)
         return
-    if os.path.isfile(foreman_ssh_authfile):
-        if foreman_ssh_key in open(foreman_ssh_authfile, 'r').read():
-            print_generic("Foreman's SSH key is already present in %s" % foreman_ssh_authfile)
+    install_ssh_key_from_string(foreman_ssh_key)
+
+
+def install_ssh_key_from_api():
+    """
+    Download and install all Foreman's SSH public keys.
+    """
+    url = "https://" + options.foreman_fqdn + ":" + str(API_PORT) + "/api/v2/smart_proxies/"
```

Comment:
This could use some apypie love (if it wasn't designed as a stand-alone script to install).